### ISSUE TITLE
feature request: more mpd functionality

### ISSUE THREAD

**Issue by bendschs** [ejurgensen/forked-daapd#533]
- send https streams to forked-daapd (at the moment it seems like only http is supported)
- set output devices with mpd-client (with client https://github.com/CDrummond/cantata this does not seem to be working, i am able to switch outputs via mpc though
- channel for rating of songs
- replay gain
- crossfading between tracks

**Comment by ejurgensen** [comment]
Thanks for the input. Do you have examples of https streams?

@chme guess this is also for you. Maybe some of it is actually already supported?

**Comment by bendschs** [comment]
My client has the ability to send https streams from youtube or soundcloud. if i run normal mpd on my linux machine i am able to send and play https streams through the api. For instance, a soundcloudlink looks like this:  https://api.soundcloud.com/tracks/29296392/stream?client_id=xxxxxxxxxxxxxxxxxxxx#{SONG_DETAILS}Artistname - Trackname<@><@>SoundCloud<@>Artist <@>Genre<@>195<@>0<@>0<@>0<@>soundcloud 

( i replaced my client id, artist-, track- and genre-info for the prupose of the example)

If i send the same, by the client generated https link to forked-daapd it tells me "unsupported protocol". So my guess is, that it does not support HTTPS.

Same behaviour with youtube or any other https-link to an audio-file.

cheers
bendsch

**Comment by chme** [comment]
@bendschs Setting rating from cantata for songs is supported with the latest release of forked-daapd (v26.0). Which version of forked-daapd are you running?

Strange that setting outputs from cantata does not work (i am not aware of another mpd client that has this problem). It looks like cantata does not like high output ids. Setting local audio or pipe outputs does work. But for airplay speakers cantata sends a wrong output id to forked-daapd.

**Comment by bendschs** [comment]
hi

I am running V 26.0

1. As for the ratings i noticed that they actually are saved correctly. while scrolling over the logs of forked-daapd i just stumbled over the notice "no such channel "ratings"". However it seems to be working.

2. Cantata actually recognizes when outputs are changed (by webinterface of forked-daapd or by mpc). In that case cantata shows a message notification "Output XY enabled/disabled". But it is not possible to change the output by Cantata Menu.

cheers
Bendsch

**Comment by bendschs** [comment]
i noticed that i can enable/disable local audio-output through cantata. it just does not work for the airplay devices.

hope that helps
bendsch

**Comment by ski522** [comment]
> * send https streams to forked-daapd (at the moment it seems like only http is supported)

Throwing my hat in the ring looking for https support as well. The following radio stream plays in MPD, but not forked-daapd. I'm finding more and more radio streams are moving to https. 

https://prod-34-203-12-125.wostreaming.net/townsquare-whomfmaac-hlsc3.m3u8

**Comment by ejurgensen** [comment]
@ski522 forked-daapd should support https now (from version 27.0), but I see the particular stream you have linked to does not work. I'm not sure why, other than ffmpeg complaining about a bunch of stuff. I haven't tried with newer versions of ffmpeg yet, perhaps then it works better.

**Comment by ski522** [comment]
> @ski522 forked-daapd should support https now (from version 27.0), but I see the particular stream you have linked to does not work. I'm not sure why, other than ffmpeg complaining about a bunch of stuff.

Yeah, I tried ffmpeg (3.4.6) which is version included with Ubuntu 18.04 distro and it didn't work, so downloaded and complied ffmpeg 4.2.1 and the stream did work with that version if this helps.
